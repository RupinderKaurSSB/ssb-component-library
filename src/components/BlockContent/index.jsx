import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { ssbGreen3, ssbWhite } from '../../style/colors';
import {Â openSans, roboto } from '../../style/mixins';

const BlockContent = ({children, pageNumber}) => (
	<Wrapper>
		<GreenLine />
		{pageNumber && <PageNumber><span>{pageNumber}</span></PageNumber>}
		{children}
	</Wrapper>
);

const Wrapper = styled.div`
	${openSans};
	background: ${ssbWhite};
	box-shadow: 0 2px 4px 1px rgba(200, 200, 200, .5);
	box-sizing: border-box;
	max-width: 1200px;
	padding: 100px;
	position: relative;
	width: 100%;
`;

const GreenLine = styled.div`
	background: ${ssbGreen3};
	height: 4px;
	left: 0;
	position: absolute;
	top: 0;
	width: 100%;
`;

const PageNumber = styled.div`
	${roboto};
	color: #667b7f;
	font-size: 30px;
	margin-top: -35px;
	
	&:before {
		background: ${ssbGreen3};
		content: '';
		display: inline-block;
		height: 22px;
		margin-right: 10px;
		width: 10px;
	}
`;

BlockContent.propTypes = {
	children: PropTypes.node.isRequired,
	pageNumber: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
};

export default BlockContent;
