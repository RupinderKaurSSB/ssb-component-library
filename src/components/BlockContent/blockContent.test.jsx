import React from 'react';
import {shallow} from 'enzyme';
import BlockContent from './index';

describe('BlockContent component', () => {
	test('Matches the snapshot', () => {
		const wrapper = shallow(<BlockContent>BlockContent</BlockContent>);
		expect(wrapper).toMatchSnapshot();
	});
	test('Toggles classNames correctly', () => {
		const wrapper = shallow(<BlockContent sectionNumber={1}>BlockContent</BlockContent>);
		expect(wrapper.find('.section-number').length).toBe(1);
	});
});
