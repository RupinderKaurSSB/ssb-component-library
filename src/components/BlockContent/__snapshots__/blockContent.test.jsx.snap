// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockContent component Matches the snapshot 1`] = `
<div
  className="ssb-block-content"
>
  BlockContent
</div>
`;

exports[`Button component Matches the snapshot 1`] = `
<div
  className="ssb-block-content"
>
  BlockContent
</div>
`;
